modules/apps/29-fee/types/fee_test.go

```diff
@@ -37,66 +49,70 @@ func TestFeeValidation(t *testing.T) {
 			func() {},
 			true,
 		},
+		{
+			"should fail when refund address is invalid",
+			func() {
+				packetFee.RefundAddress = "invalid-address"
+			},
+			false,
+		},
 		{
 			"should fail when all fees are invalid",
 			func() {
-				ackFee = invalidCoins
-				receiveFee = invalidCoins
-				timeoutFee = invalidCoins
+				packetFee.Fee.AckFee = invalidFee
+				packetFee.Fee.RecvFee = invalidFee
+				packetFee.Fee.TimeoutFee = invalidFee
 			},
 			false,
 		},
 		{
 			"should fail with single invalid fee",
 			func() {
-				ackFee = invalidCoins
+				packetFee.Fee.AckFee = invalidFee
 			},
 			false,
 		},
 		{
 			"should fail with two invalid fees",
 			func() {
-				timeoutFee = invalidCoins
-				ackFee = invalidCoins
+				packetFee.Fee.TimeoutFee = invalidFee
+				packetFee.Fee.AckFee = invalidFee
 			},
 			false,
 		},
 		{
 			"should pass with two empty fees",
 			func() {
-				timeoutFee = sdk.Coins{}
-				ackFee = sdk.Coins{}
+				packetFee.Fee.TimeoutFee = sdk.Coins{}
+				packetFee.Fee.AckFee = sdk.Coins{}
 			},
 			true,
 		},
 		{
 			"should pass with one empty fee",
 			func() {
-				timeoutFee = sdk.Coins{}
+				packetFee.Fee.TimeoutFee = sdk.Coins{}
 			},
 			true,
 		},
 		{
 			"should fail if all fees are empty",
 			func() {
-				ackFee = sdk.Coins{}
-				receiveFee = sdk.Coins{}
-				timeoutFee = sdk.Coins{}
+				packetFee.Fee.AckFee = sdk.Coins{}
+				packetFee.Fee.RecvFee = sdk.Coins{}
+				packetFee.Fee.TimeoutFee = sdk.Coins{}
 			},
 			false,
 		},
 	}
 
 	for _, tc := range testCases {
-		// build message
-		ackFee = validCoins
-		receiveFee = validCoins
-		timeoutFee = validCoins
-
-		// malleate
-		tc.malleate()
-		fee = Fee{receiveFee, ackFee, timeoutFee}
-		err := fee.Validate()
+		fee := types.NewFee(defaultRecvFee, defaultAckFee, defaultTimeoutFee)
+		packetFee = types.NewPacketFee(fee, defaultAccAddress, nil)
```

Comment:
Why don't you create a `defaultPacketFee`

Reply:
This `packetFee` is mutated by `malleate()` in each of the testcases, so creating a `defaultPacketFee` as a global test variable might not be great for reusing in other tests, as whatever its state is after the last case would carry forward to the next test. So you'd have to be careful resetting it..etc 

I think here, because the `packetFee` is the subject under test it makes more sense to not be using a global var. The fields defined globally are fairly "static" for lack of a better word and more so boilerplate for not over bloating the test cases with initialisation code.